api/include/opentelemetry/trace/noop.h

```diff
@@ -66,7 +66,12 @@ class NoopTracer final : public Tracer, public std::enable_shared_from_this<Noop
                                     const KeyValueIterable & /*attributes*/,
                                     const StartSpanOptions & /*options*/) noexcept override
   {
-    return nostd::shared_ptr<Span>{new (std::nothrow) NoopSpan{this->shared_from_this()}};
+    // Don't allocate a no-op span for every StartSpan call, but use a static
+    // singleton for this case.
+    static nostd::shared_ptr<trace_api::Span> noop_span(
+        new trace_api::NoopSpan{this->shared_from_this()});
+
+    return noop_span;
```

Comment:
Wondering whether there could be issue if the caller releases the returned `shared_ptr`.

Reply:
It seems ref-count is not an issue here. Could we use `static auto noop_span(std::make_shared<trace_api::NoopSpan>(this->shared_from_this()))` which could save 1 heap allocation (the span and shared_ptr will be allocated in one heap allocation call).

Reply:
We need a `nostd::shared_ptr` here, not a `std::shared_ptr`. So I'm afraid using `make_shared` will not benefit us here.

Reply:
Ok. It makes sense if `nostd::shared_ptr` has to be used here.